src/_pytest/pytester.py

```diff
@@ -634,46 +661,48 @@ def __init__(self, request: FixtureRequest, tmpdir_factory: TempdirFactory) -> N
         else:
             name = request.node.name
         self._name = name
-        self.tmpdir = tmpdir_factory.mktemp(name, numbered=True)
-        self.test_tmproot = tmpdir_factory.mktemp("tmp-" + name, numbered=True)
+        self._path: Path = tmp_path_factory.mktemp(name, numbered=True)
         self.plugins: List[Union[str, _PluggyPlugin]] = []
         self._cwd_snapshot = CwdSnapshot()
         self._sys_path_snapshot = SysPathsSnapshot()
         self._sys_modules_snapshot = self.__take_sys_modules_snapshot()
         self.chdir()
-        self.request.addfinalizer(self.finalize)
-        self._method = self.request.config.getoption("--runpytest")
+        self._request.addfinalizer(self._finalize)
+        self._method = self._request.config.getoption("--runpytest")
 
-        mp = self.monkeypatch = MonkeyPatch()
-        mp.setenv("PYTEST_DEBUG_TEMPROOT", str(self.test_tmproot))
+        self._monkeypatch = mp = MonkeyPatch()
+        mp.setenv("PYTEST_DEBUG_TEMPROOT", str(tmp_path_factory.mktemp("tmproot")))
```

Comment:
The move of tmpdir into the tester root is potentially a breaking change and might break testsuites

Reply:
True, but I suppose this will be very rare and if it happens, the fix should be simple, so I think that's OK .

Reply:
To avoid any breakages I reverted to the old behavior, but I did keep `test_tmproot` private in `Pytester`.

Do you remember the purpose of that variable @RonnyPfannschmidt?